Tests fail when TMPDIR contains hyphens

We've noticed that the test suite fails on Gentoo because the `TMPDIR` used while building packages contains hyphens. It seems that sometimes substitutes all hyphens in path with underscores.

To reproduce:

```
export TMPDIR=/tmp/cheetah3-3.3.1/temp
mkdir -p "${TMPDIR}"
tox
```

<details>
<summary>Error output</summary>

```pytb
.pkg: _optional_hooks> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
py310: install_package> python -I -m pip install --force-reinstall --no-deps /tmp/cheetah3-3.3.1/.tox/.tmp/package/2/CT3-3.3.1.tar.gz
py310: commands[0]> .tox/py310/bin/python --version
Python 3.10.10
py310: commands[1]> .tox/py310/bin/python -c 'import struct; print(struct.calcsize('"'"'P'"'"') * 8)'
64
py310: commands[2]> .tox/py310/bin/python /tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Test.py
.........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................E.EEEE.FF...................................................
======================================================================
ERROR: runTest (Cheetah.Tests.Unicode.JBQ_UTF8_Test7)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 131, in runTest
    template = self.createAndCompile(source)
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 24, in createAndCompile
    fd = open('%s.tmpl' % sourcefile, 'w', encoding='utf-8')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/cheetah3_3.3.1/temp/tmplre8dnao.tmpl'

======================================================================
ERROR: testStaticCompile (Cheetah.Tests.Unicode.JBQ_UTF8_Test8)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 143, in testStaticCompile
    template = self.createAndCompile(source)()
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 24, in createAndCompile
    fd = open('%s.tmpl' % sourcefile, 'w', encoding='utf-8')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/cheetah3_3.3.1/temp/tmpcnzh_309.tmpl'

======================================================================
ERROR: test_BasicASCII (Cheetah.Tests.Unicode.Unicode_in_SearchList_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 185, in test_BasicASCII
    template = self.createAndCompile(source)
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 24, in createAndCompile
    fd = open('%s.tmpl' % sourcefile, 'w', encoding='utf-8')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/cheetah3_3.3.1/temp/tmplkrv57du.tmpl'

======================================================================
ERROR: test_Thai (Cheetah.Tests.Unicode.Unicode_in_SearchList_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 193, in test_Thai
    template = self.createAndCompile(source)
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 24, in createAndCompile
    fd = open('%s.tmpl' % sourcefile, 'w', encoding='utf-8')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/cheetah3_3.3.1/temp/tmp3z63xypm.tmpl'

======================================================================
ERROR: test_Thai_utf8 (Cheetah.Tests.Unicode.Unicode_in_SearchList_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 210, in test_Thai_utf8
    template = self.createAndCompile(source)
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Unicode.py", line 24, in createAndCompile
    fd = open('%s.tmpl' % sourcefile, 'w', encoding='utf-8')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/cheetah3_3.3.1/temp/tmpufwexe2x.tmpl'

======================================================================
FAIL: testFill (Cheetah.Tests.CheetahWrapper.FileDataFill)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/CheetahWrapper.py", line 628, in testFill
    self.go('cheetah fill --json {jsonfile} json.tmpl'
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/CheetahWrapper.py", line 237, in go
    output = self.assertSubprocess(cmd)
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/CheetahWrapper.py", line 217, in assertSubprocess
    self.assertEqual(status, 0,
AssertionError: 1 != 0 : Subprocess exited with a non-zero status (1)
                             b'Traceback (most recent call last):\n  File "/tmp/cheetah3-3.3.1/.tox/py310/bin/cheetah", line 3, in <module>\n    _cheetah()\n  File "/tmp/cheetah3-3.3.1/Cheetah/CheetahWrapper.py", line 664, in _cheetah\n    CheetahWrapper().main()\n  File "/tmp/cheetah3-3.3.1/Cheetah/CheetahWrapper.py", line 132, in main\n    self.parseOpts(argv[2:])\n  File "/tmp/cheetah3-3.3.1/Cheetah/CheetahWrapper.py", line 257, in parseOpts\n    f = open(opts.json, \'r\')\nFileNotFoundError: [Errno 2] No such file or directory: \'/tmp//tmp/cheetah3-3.3.1/.tox/py310/bin/cheetah3-3.3.1/temp/tmp51ab6_bf\'\n'

======================================================================
FAIL: testText (Cheetah.Tests.CheetahWrapper.FileDataFill)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/CheetahWrapper.py", line 636, in testText
    self.go('cheetah fill --json {jsonfile} --oext txt json.tmpl'
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/CheetahWrapper.py", line 237, in go
    output = self.assertSubprocess(cmd)
  File "/tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/CheetahWrapper.py", line 217, in assertSubprocess
    self.assertEqual(status, 0,
AssertionError: 1 != 0 : Subprocess exited with a non-zero status (1)
                             b'Traceback (most recent call last):\n  File "/tmp/cheetah3-3.3.1/.tox/py310/bin/cheetah", line 3, in <module>\n    _cheetah()\n  File "/tmp/cheetah3-3.3.1/Cheetah/CheetahWrapper.py", line 664, in _cheetah\n    CheetahWrapper().main()\n  File "/tmp/cheetah3-3.3.1/Cheetah/CheetahWrapper.py", line 132, in main\n    self.parseOpts(argv[2:])\n  File "/tmp/cheetah3-3.3.1/Cheetah/CheetahWrapper.py", line 257, in parseOpts\n    f = open(opts.json, \'r\')\nFileNotFoundError: [Errno 2] No such file or directory: \'/tmp//tmp/cheetah3-3.3.1/.tox/py310/bin/cheetah3-3.3.1/temp/tmp_5ukkv4z\'\n'

----------------------------------------------------------------------
Ran 2181 tests in 10.025s

FAILED (failures=2, errors=5)
py310: exit 1 (10.37 seconds) /tmp/cheetah3-3.3.1> .tox/py310/bin/python /tmp/cheetah3-3.3.1/.tox/py310/lib/python3.10/site-packages/Cheetah/Tests/Test.py pid=2004668
.pkg: _exit> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
  py310: FAIL code 1 (17.91=setup[7.52]+cmd[0.00,0.02,10.37] seconds)
  evaluation failed :( (18.00 seconds)

```
</details>

--- a/Cheetah/Tests/Unicode.py
+++ b/Cheetah/Tests/Unicode.py
@@ -1,6 +1,5 @@
 # -*- encoding: utf8 -*-
 
-from glob import glob
 import os
 from shutil import rmtree
 import tempfile
@@ -13,33 +12,28 @@ from Cheetah.compat import PY2, unicode, load_module_from_file
 
 class CommandLineTest(unittest.TestCase):
     def createAndCompile(self, source):
-        fd, sourcefile = tempfile.mkstemp()
-        os.close(fd)
-        os.remove(sourcefile)
-        sourcefile = sourcefile.replace('-', '_')
+        tmpDir = tempfile.mkdtemp()
+        try:
+            sourcefile = os.path.join(tmpDir, "source")
 
-        if PY2:
-            fd = open('%s.tmpl' % sourcefile, 'w')
-        else:
-            fd = open('%s.tmpl' % sourcefile, 'w', encoding='utf-8')
-        fd.write(source)
-        fd.close()
+            if PY2:
+                fd = open('%s.tmpl' % sourcefile, 'w')
+            else:
+                fd = open('%s.tmpl' % sourcefile, 'w', encoding='utf-8')
+            fd.write(source)
+            fd.close()
 
-        wrap = CheetahWrapper.CheetahWrapper()
-        wrap.main(['cheetah', 'compile',
-                   '--encoding=utf-8', '--settings=encoding="utf-8"',
-                   '--quiet', '--nobackup', sourcefile])
-        module_name = os.path.split(sourcefile)[1]
-        module = load_module_from_file(
-            module_name, module_name, sourcefile + '.py')
-        template = getattr(module, module_name)
-        os.remove('%s.tmpl' % sourcefile)
-        for sourcefile_py in glob('%s.py*' % sourcefile):  # *.py[co]
-            os.remove(sourcefile_py)
-        __pycache__ = os.path.join(os.path.dirname(sourcefile), '__pycache__')
-        if os.path.exists(__pycache__):  # PY3
-            rmtree(__pycache__)
-        return template
+            wrap = CheetahWrapper.CheetahWrapper()
+            wrap.main(['cheetah', 'compile',
+                       '--encoding=utf-8', '--settings=encoding="utf-8"',
+                       '--quiet', '--nobackup', sourcefile])
+            module_name = os.path.split(sourcefile)[1]
+            module = load_module_from_file(
+                module_name, module_name, sourcefile + '.py')
+            template = getattr(module, module_name)
+            return template
+        finally:
+            rmtree(tmpDir, True)
 
 
 class JBQ_UTF8_Test1(unittest.TestCase):

